Snowflake.Data/Core/SFSessionProperty.cs

```diff
@@ -129,101 +131,21 @@ public override int GetHashCode()
-                                    singleEqualIndex = -1;
-                                    break;
-                                }
-                            }
-                            else
-                            {
-                                // skip the doubled one
-                                currentIndex += 2;
-                            }
-                        }
-
-                        if ((singleEqualIndex > 0) && (singleEqualIndex < keyVal.Length - 1))
-                        {
-                            // Split the key/value at the right index and deduplicate '=='
-                            tokens = new string[2];
-                            tokens[0] = keyVal.Substring(0, singleEqualIndex).Replace("==", "=");
-                            tokens[1] = keyVal.Substring(
-                                singleEqualIndex + 1,
-                                keyVal.Length - (singleEqualIndex + 1)).Replace("==", "="); ;
-                        }
-                        else
-                        {
-                            // An equal sign was not doubled or something else happened
-                            // making the connection invalid
-                            string invalidStringDetail =
-                                String.Format("Invalid key value pair {0}", keyVal);
-                            SnowflakeDbException e =
-                                new SnowflakeDbException(
-                                    SFError.INVALID_CONNECTION_STRING,
-                                    new object[] { invalidStringDetail });
-                            logger.Error("Invalid string.", e);
-                            throw e;
-                        }
-                    }
-
-                    // temporary change to pretend as ODBC to enable multiple
-                    // statements on server side, needs to be removed when merge.
-                    if (tokens[0].ToLower() == "fakeodbc")
-                    {
-                        if (tokens[1].ToLower() == "true")
-                        {
-                            SFEnvironment.DriverName = "ODBC";
-                            SFEnvironment.DriverVersion = "2.25.2";
-                        }
-                        continue;
-                    }
-                    try
-                    {
-                        SFSessionProperty p = (SFSessionProperty)Enum.Parse(
-                            typeof(SFSessionProperty), tokens[0].ToUpper());
-                        properties.Add(p, tokens[1]);
-                    }
-                    catch (ArgumentException e)
-                    {
-                        logger.Warn($"Property {tokens[0]} not found ignored.", e);
-                    }
-                }
+                SFSessionProperty p = (SFSessionProperty)Enum.Parse(
+                            typeof(SFSessionProperty), keys[i].ToUpper());
+                properties.Add(p, values[i]);

```

Comment:
Check "fakeodbc" case above. It was ignore. Also when it's true it was setting SFEnvironment stuff. You need to do the same with your new logic here.